src/idlechecker/idle-checker.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# idle-checker.sh checks for cpu utilization of the VM and update last active timestamp if the CPU utilization
+# is above a given threshold.
+#
+# Note that this script is dependent on some variables and packages being set up:
+#
+# - AWS CLI is installed
+# - CLOUD: cloud platform VM is running on
+# - bc https://manpages.ubuntu.com/manpages/trusty/en/man1/bc.1.html
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+# threshold - threshold for cpu usage, used to determine if instance is idle. If usage goes above this number count resets to zero. By default 0.1 (10 percent)
+readonly threshold="${THRESHOLD:-0.1}"
+readonly LAST_ACTIVE_KEY="last-active/cpu"
+
+function set_guest_attributes() {
+  echo "Setting cpu-utilization/last-active"
+  curl -s -X PUT --data "$1" \
+    -H "Metadata-Flavor: Google" \
+    "http://metadata.google.internal/computeMetadata/v1/instance/guest-attributes/${LAST_ACTIVE_KEY}"
+}
+readonly -f set_guest_attributes
+
+function create_tag() {
+  echo "Creating tag vwbapp:cpu-utilization/last-active"
+  INSTANCE_ID="$(wget -q -O - http://169.254.169.254/latest/meta-data/instance-id)"
```

Comment:
```suggestion
  local INSTANCE_ID
  INSTANCE_ID="$(wget -q -O - http://169.254.169.254/latest/meta-data/instance-id)"
```

I also have a preference for local variables to be lower-cased, so even better would be:

```suggestion
  local id
  id="$(wget -q -O - http://169.254.169.254/latest/meta-data/instance-id)"
```